pkgs/native_assets_cli/lib/src/config.dart

```diff
@@ -207,6 +186,10 @@ final class BuildConfigBuilder extends HookConfigBuilder {
     json[_dependencyMetadataKey] = {
       for (final key in metadata.keys) key: metadata[key]!.toJson(),
     };
+    // TODO: Bump min-SDK constraint to 3.7 and remove once stable.
```

Comment:
This function should no longer take `dryRun` instead it should hardcode it to `false` in the json.

Reply:
That would break the tests in `native_toolchain_c` that check that no assets are built if dryRun is true. We can only remove such functionality once we can bump the Dart SDK constraint to 3.7 stable (assuming we get all the refactorings landed in both Dart and Flutter before the branchcut for 3.7).

Reply:
This code here is the builder code (i.e. code for Dart SDK & Flutter SDK - which no longer set this to `true`), no?

The `package:native_toolchain_c` is for hook writers, it can still read the dry run (possibly produced by older versions of Dart SDK & older versions of Flutter SDK).

Reply:
The builder code is used to produce a config in the unit tests of `native_toolchain_c`.